cmd/api/src/api/v2/apiclient/ingest.go

```diff
@@ -71,6 +75,39 @@ func (s Client) SendFileUploadData(payload map[string]any, id int64) error {
 	}
 }
 
+func (s Client) SendCompressedFileUploadData(jsonFile io.Reader, id int64) error {
+	var (
+		body bytes.Buffer
+		gw   = gzip.NewWriter(&body)
+	)
+
+	if n, err := io.Copy(gw, jsonFile); err != nil {
+		return fmt.Errorf("failed to write compressed json data %w", err)
+	} else if n == 0 {
+		return errors.New("zero bytes written to compressed body payload")
+	} else if err = gw.Close(); err != nil {
+		return fmt.Errorf("failed to close gzip writer: %w", err)
+	}
+
+	request, err := s.NewRequest(http.MethodPost, fmt.Sprintf("api/v2/file-upload/%d", id), nil, io.NopCloser(&body))
+	if err != nil {
+		return fmt.Errorf("failed to create compressed ingest request: %w", err)
+	}
+	request.Header.Set("Content-Encoding", "gzip")
```

Comment:
nit: encouraging the use of `headers.ContentEncoding.String()` to avoid potential typos.

Reply:
Done!

Reply:
Good catch, I found the other places and fixed them